common/src/main/java/com/sedmelluq/discord/lavaplayer/natives/NativeLibLoader.java

```diff
@@ -124,37 +189,65 @@ private static void createDirectoriesWithFullPermissions(Path path) throws IOExc
     }
   }
 
-  private static Architecture detectArchitecture() {
-    String osName = System.getProperty("os.name");
-    String arch = System.getProperty("os.arch");
-    String bits = System.getProperty("sun.arch.data.model");
+  private static class Architecture {
+    private final String osName;
+    private final String archName;
+    private final String libraryPrefix;
+    private final String librarySuffix;
+    private final String systemType;
 
-    if (osName.startsWith("Linux")) {
-      if (arch.startsWith("arm")) {
-        return new Architecture(LINUX_ARM, "lib", ".so");
-      } else if (arch.startsWith("aarch64")) {
-        return new Architecture(LINUX_ARM_64, "lib", ".so");
+    private static String formatSystemName(String osName, String archName) {
+      if (osName == OS_OSX && archName == ARCH_X86_64) {
+        return OS_DARWIN;
       } else {
-        return new Architecture("64".equals(bits) ? LINUX_X86_64 : LINUX_X86, "lib", ".so");
+        return osName + "-" + archName;
       }
-    } else if ((osName.startsWith("Mac") || osName.startsWith("Darwin")) && "64".equals(bits)) {
-      return new Architecture(DARWIN, "lib", ".dylib");
-    } else if (osName.startsWith("Windows") && !osName.startsWith("Windows CE")) {
-      return new Architecture("64".equals(bits) ? WIN_X86_64 : WIN_X86, "", ".dll");
     }
 
-    throw new IllegalStateException("Native libraries not supported on this platform.");
-  }
+    public String formatLibraryName(String libName) {
+      return this.libraryPrefix + libName + this.librarySuffix;
+    }
 
-  private static class Architecture {
-    private final String systemType;
-    private final String libraryPrefix;
-    private final String librarySuffix;
+    public static Architecture detectArchitecture() throws IllegalStateException, IllegalArgumentException {
+      final String _os_arch = System.getProperty("os.arch");
+      final String _os_name = System.getProperty("os.name");
+      String archName = System.getProperty("lavaplayer.native.arch", archMap.get(_os_arch));
+      String osName, libPrefix, libSuffix;
+
+      if (archName == null) {
+        throw new IllegalArgumentException("Unknown architecture: " + _os_arch);
```

Comment:
No arch detection should be performed at all if `lavaplayer.native.system` is specified. Actually the `archName` field is not even used, so could pass only `systemType` into the constructor.

Reply:
Additionally, if OS is undetected, but `lavaplayer.native.system` is specified, it might make sense to use Linux as the base (for prefix, suffix) as it's most likely some flavor of unix which probably has the same library name conventions.